client/config.go

```diff
@@ -0,0 +1,153 @@
+package client
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+
+	as "github.com/aerospike/aerospike-client-go/v6"
+)
+
+type AerospikeConfig struct {
+	Seeds                  HostTLSPortSlice
+	User                   string
+	Password               string
+	AuthMode               as.AuthMode
+	RootCA                 [][]byte
+	Cert                   []byte
+	Key                    []byte
+	KeyPass                []byte
+	TLSProtocolsMinVersion TLSProtocol
+	TLSProtocolsMaxVersion TLSProtocol
+	// TLSCipherSuites        []uint16 // TODO
+}
+
+func NewDefaultAerospikeHostConfig() *AerospikeConfig {
+	return &AerospikeConfig{
+		Seeds: HostTLSPortSlice{NewDefaultHostTLSPort()},
+	}
+}
+
+func (ac *AerospikeConfig) NewClientPolicy() (*as.ClientPolicy, error) {
+	clientPolicy := as.NewClientPolicy()
+	clientPolicy.User = ac.User
+	clientPolicy.Password = ac.Password
+	clientPolicy.AuthMode = ac.AuthMode
+
+	tlsConfig, err := ac.newTLSConfig()
+
+	if err != nil {
+		return nil, err
+	}
+
+	clientPolicy.TlsConfig = tlsConfig
+
+	return clientPolicy, nil
+}
+
+func (ac *AerospikeConfig) NewHosts() []*as.Host {
+	hosts := []*as.Host{}
+
+	for _, seed := range ac.Seeds {
+		host := as.NewHost(seed.Host, seed.Port)
+
+		if seed.TLSName != "" {
+			host.TLSName = seed.TLSName
+		}
+
+		hosts = append(hosts, host)
+	}
+
+	return hosts
+}
+
+func (ac *AerospikeConfig) newTLSConfig() (*tls.Config, error) {
+	if len(ac.RootCA) == 0 && len(ac.Cert) == 0 && len(ac.Key) == 0 {
+		return nil, nil
+	}
+
+	var (
+		clientPool []tls.Certificate
+		serverPool *x509.CertPool
+		err        error
+	)
+
+	serverPool = loadCACerts(ac.RootCA)
+
+	if len(ac.Cert) > 0 || len(ac.Key) > 0 {
+		clientPool, err = loadServerCertAndKey(ac.Cert, ac.Key, ac.KeyPass)
+		if err != nil {
+			return nil, fmt.Errorf("failed to load server certificate and key: `%s`", err)
+		}
```

Comment:
This might be something carried over from the exporter? My understanding is that in a deployment the cert and key would be owned by the server. When a client wants to connect to the server, the client needs to allow the server to authenticate the client's identity (only in mutual auth). So in this case these are certs and keys installed so that the server can authenticate. These are _for_ the server's benefit. Which is different from the RootCA which is _for_ the client's benefit

Reply:
Let me know if you think I should just remove the word "server"

Reply:
I think "Cert" is a little ambiguous in these cases

Reply:
That explanation makes sense to me but I think you should remove the word "server" from the error message. What about this? 
```suggestion
		clientPool, err = loadServerCertAndKey(ac.Cert, ac.Key, ac.KeyPass)
		if err != nil {
			return nil, fmt.Errorf("failed to load client authentication certificate and key `%s`", err)
		}
```
The go docs mention "client-authentication" for the Certificates field of the tls config this goes into https://pkg.go.dev/crypto/tls#Config